### PR TITLE
Generalize filter component and create story translations filter 

### PR DESCRIPTION
**Pull request by andrewyguo**
## Notion ticket link

[Create Story Translations filters](https://www.notion.so/uwblueprintexecs/Create-Story-Translations-filters-263b6b271ede47bda3b1fc7bbb506b9e)

## Implementation description

- Generalize the filter component so that it can be used for both the User table and Story Translation table 
- Modify `buildStoriesQuery` in `frontend\src\APIClients\queries\StoryQueries.ts` to take in filter parameters 

## Steps to test

1. Go to the Manage  Story Translations section of the admin page and make sure filtered queries return as expected 
2. Go to the Manage Translators and Manage Reviewers page as well to make sure nothing has broken after the filter component has been generalized 

## What should reviewers focus on?

- None of the filters are broken 
- Try some different combinations of filters 

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/APIClients/queries/StoryQueries.ts [hunk 1]
```diff
@@ -137,13 +137,23 @@ export type StoryTranslation = {
...
+  queryParams += stage ? `stage: "${stage}", ` : "";
+  queryParams += storyTitle ? `storyTitle: "${storyTitle}", ` : "";
+  queryParams = queryParams === "" ? queryParams : `( ${queryParams} )`;
+
   return {
     fieldName: "storyTranslations",
     string: gql`
       query StoryTranslations {
-        storyTranslations {
+        storyTranslations ${queryParams} {
```
Comment (andrewyguo) [inline frontend/src/APIClients/queries/StoryQueries.ts:156]: Not enclosed in parenthesis because if we have an empty set of `queryParams` and create a query like `storyTranslations () { ... } ` GQL throws an error

File: backend/python/app/services/implementations/story_service.py [hunk 2]
```diff
@@ -145,6 +148,8 @@ def get_story_translations(
                 filters.append(Story.level == level)
             if story_title is not None:
                 filters.append(Story.title.like(f"%{story_title}%"))
+            if role_filter is not None:
+                filters.append(role_filter)
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:152]: out of curiosity, why the change in order of applying filters?

File: frontend/src/APIClients/queries/StoryQueries.ts (hunk 1 above)
Comment (gaoxk) [reply frontend/src/APIClients/queries/StoryQueries.ts:161]: lgtm 👍  lines 151 - 155 are pretty satisfying lookin 😍

**Review by gaoxk — APPROVED** [review]
noicee

File: backend/python/app/services/implementations/story_service.py (hunk 2 above)
Comment (andrewyguo) [reply backend/python/app/services/implementations/story_service.py:152]: LOL so I had initially removed the role_filter filter while debugging a GQL error but I realized that that wasn't what was causing it and added it back in. I just followed the order in which the parameters were listed above.
